### ISSUE TITLE
Review uses of `std::pair`

### ISSUE THREAD

**Issue by DanRStevens** [lairworks/nas2d-core#399]
We should review uses of `std::pair` to see if `Point` or `Vector` may be a better alternative.

A quick `grep` shows these areas of interest:
```
include/NAS2D/Resources/Image.h:67:	std::pair<int, int> _size{0,0};		   /**< Width/Height information about the Image. */
include/NAS2D/Resources/Image.h:68:	std::pair<uint32_t, uint32_t> _center{0u,0u}; /**<  */
include/NAS2D/StringUtils.h:25:	std::pair<std::string, std::string> splitOnFirst(const std::string& str, char delim);
include/NAS2D/StringUtils.h:26:	std::pair<std::string, std::string> splitOnLast(const std::string& str, char delim);
include/NAS2D/Xml/XmlBase.h:118:	std::pair<int, int> location;
include/NAS2D/Xml/XmlDocument.h:58:	std::pair<int, int>	_errorLocation;
src/StringUtils.cpp:85:std::pair<std::string, std::string> NAS2D::splitOnFirst(const std::string& str, char delim)
src/StringUtils.cpp:98:std::pair<std::string, std::string> NAS2D::splitOnLast(const std::string& str, char delim)
src/Xml/XmlParser.cpp:105:	const std::pair<int, int>& cursor() const { return _cursor; }
src/Xml/XmlParser.cpp:115:	std::pair<int, int>	_cursor;
```

**Comment by cugone** [comment]
Closed via #461 and #462

**Comment by DanRStevens** [comment]
Small point, but you should generally wait until after changes have been merged before closing.
